analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("s1", "s3"), LiveOut("s1", "s3"));
+        context.Validate(context.Cfg.Blocks[9], null, LiveIn("s1", "s3"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_Overwrite()
+    {
+        /*       Block 1
+         *       #0=s1
+         *         |
+         *       Block 2
+         *   #1=(s1="overwrite")
+         *     #1 is null
+         *        /   \
+         *       T     F
+         *       /     \
+         *    Block3   Block4
+         *    #2=#1   #2="value"
+         *      \         /
+         *       \       /
+         *        Block5
+         *        s1=#2
+         */
+        const string code = """s1 = (s1 = "overwrite")  ?? "value";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1");
+        context.ValidateEntry(LiveIn("s1"), LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1"));
+        context.Validate(context.Cfg.Blocks[2], null);
+        context.Validate(context.Cfg.Blocks[3], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_SwitchStatement()
+    {
+        const string code = """
+            string result = i switch
+            {
+                0 =>  stringParam,
+                1 => "Something",
+                _ => "Everything"
+            };
+            """;
+        var context = CreateContextCS(code, additionalParameters: "int i, string stringParam");
+        context.ValidateEntry(LiveIn("i", "stringParam"), LiveOut("i", "stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 0
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = stringParam
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 1
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = "Something"
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("i", "stringParam"), LiveOut("stringParam"));      // if i != 0 && i != 1
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = "Everything"
+        context.Validate(context.Cfg.Blocks[7], null);                                                          // to no destination block
```

Comment:
The block numbers doesn't seem to match the description. 7 is "Everything", 8 is unreachable throw and 9 is missing with the result

Reply:
I'm a bit confused. To me 7 seems to be unreachable and 9 is the exit.
Here is the CFG:

```
 digraph "RoslynCfg" {
subgraph "cluster_1" {
label = "LocalLifetime region, Locals: result, Captures: #Capture-0"
subgraph "cluster_2" {
label = "LocalLifetime region, Captures: #Capture-1"
cfg0_block1 [shape=record label="{BLOCK #1|0#: FlowCaptureOperation: #Capture-1: i|1#: 0#.Value: ParameterReferenceOperation: i|##########|## BranchValue ##|0#: IsPatternOperation: 0 =\>  param|1#: 0#.Value: FlowCaptureReferenceOperation: #Capture-1: i|1#: 0#.Pattern: ConstantPatternOperation: 0|2#: 1#.Value: LiteralOperation: 0|##########}"]
cfg0_block2 [shape=record label="{BLOCK #2|0#: FlowCaptureOperation: #Capture-0: param|1#: 0#.Value: ParameterReferenceOperation: param|##########}"]
cfg0_block3 [shape=record label="{BLOCK #3|## BranchValue ##|0#: IsPatternOperation: 1 =\> \"Something\"|1#: 0#.Value: FlowCaptureReferenceOperation: #Capture-1: i|1#: 0#.Pattern: ConstantPatternOperation: 1|2#: 1#.Value: LiteralOperation: 1|##########}"]
cfg0_block4 [shape=record label="{BLOCK #4|0#: FlowCaptureOperation: #Capture-0: \"Something\"|1#: 0#.Value: LiteralOperation: \"Something\"|##########}"]
cfg0_block5 [shape=record label="{BLOCK #5|## BranchValue ##|0#: IsPatternOperation: _ =\> \"Everything\"|1#: 0#.Value: FlowCaptureReferenceOperation: #Capture-1: i|1#: 0#.Pattern: DiscardPatternOperation: _|##########}"]
cfg0_block6 [shape=record label="{BLOCK #6|0#: FlowCaptureOperation: #Capture-0: \"Everything\"|1#: 0#.Value: LiteralOperation: \"Everything\"|##########}"]
}
cfg0_block7 [shape=record label="{BLOCK #7|## BranchValue ##|0#: ObjectCreationOperation: i switch\n\{\n    0 =\>  param,\n    1 =\> \"Something\",\n    _ =\> \"Everything\"\n\}|##########}"]
cfg0_block8 [shape=record label="{BLOCK #8|0#: SimpleAssignmentOperation: result = i switch\n\{\n    0 =\>  param,\n    1 =\> \"Something\",\n    _ =\> \"Everything\"\n\}|1#: 0#.Target: LocalReferenceOperation: result = i switch\n\{\n    0 =\>  param,\n    1 =\> \"Something\",\n    _ =\> \"Everything\"\n\}|1#: 0#.Value: FlowCaptureReferenceOperation: #Capture-0: i switch\n\{\n    0 =\>  param,\n    1 =\> \"Something\",\n    _ =\> \"Everything\"\n\}|##########}"]
}
cfg0_block0 [shape=record label="{ENTRY #0}"]
cfg0_block9 [shape=record label="{EXIT #9}"]
cfg0_block0 -> cfg0_block1
cfg0_block1 -> cfg0_block2 [label="Else"]
cfg0_block1 -> cfg0_block3 [label="WhenFalse"]
cfg0_block3 -> cfg0_block4 [label="Else"]
cfg0_block3 -> cfg0_block5 [label="WhenFalse"]
cfg0_block5 -> cfg0_block6 [label="Else"]
cfg0_block5 -> cfg0_block7 [label="WhenFalse"]
cfg0_block7 -> NoDestination_cfg0_block7 [label="Throw"]
cfg0_block2 -> cfg0_block8
cfg0_block4 -> cfg0_block8
cfg0_block6 -> cfg0_block8
cfg0_block8 -> cfg0_block9
}
```

Reply:
The CFG visualizer is probably lying